cmd/link.go

```diff
@@ -2,72 +2,130 @@ package cmd
 
 import (
 	"fmt"
+	"github.com/golang/glog"
 	"github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/secret"
 	"os"
 
 	"github.com/redhat-developer/odo/pkg/component"
 	svc "github.com/redhat-developer/odo/pkg/service"
 	"github.com/spf13/cobra"
 )
 
+var (
+	port string
+)
+
 var linkCmd = &cobra.Command{
-	Use:   "link <service> --component [component]",
-	Short: "Link component to a service",
-	Long: `Link component to a service
+	Use:   "link <service> --component [component] OR link <component> --component [component]",
+	Short: "Link component to a service or component",
+	Long: `Link component to a service or component
+
+If the source component is not provided, the current active component is assumed.
+
+In both use cases, link adds the appropriate secret (from either the target component or service)
+to the environment of the source component.
+The source component can then consume the entries of the secret as environment variables 
+(note that the values of the environment variables have already been base64 decoded by the platform).
+
+An example of linking components is the following:
+We have created a backend application called 'backend' with port 8080 exposed.
+Then a frontend application is created and linked to the backend via:
+odo link backend
+
+Now frontend has 2 environment variables it can use:
+COMPONENT_BACKEND_HOST=backend-app
+COMPONENT_BACKEND_PORT=8080
 
-If source component is not provided, the link is created to the current active
-component.
+An example of linking a component to a service is the following:
+We have created a backend application called 'backend'
+Then a service was created for example via:
+odo service create dh-postgresql-apb --plan dev -p postgresql_user=luke -p postgresql_password=secret
 
-During the linking process, the secret that is created during the service creation (odo service create),
-is injected into the component.
+And linke using:
```

Comment:
link

Reply:
see review comments:

```sh
If the source component is not provided, the current active component is assumed.

In both use cases, link adds the appropriate secret to the environment of the source component. The source component can then consume the entries of the se
cret as environment variables.

For example:

We have created a backend application called 'backend' with port 8080 exposed:
odo create backend nodejs --port 8080

We've also created a frontend application called 'frontend':
odo create frontend nodejs

You can now link the two applications:
odo link backend --component frontend

Now the frontend has 2 ENV variables it can use:
COMPONENT_BACKEND_HOST=backend-app
COMPONENT_BACKEND_PORT=8080

If you wish to use a database, we can use the Service Catalog and link it to our backend:
odo service create dh-postgresql-apb --plan dev -p postgresql_user=luke -p postgresql_password=secret
odo link dh-postgresql-apb

Now backend has 2 ENV variables it can use:
DB_USER=luke
DB_PASSWORD=secret
```

Reply:
Done